mapiproxy/libmapiproxy/backends/openchangedb_ldb.c

```diff
@@ -1772,6 +1782,164 @@ static const char **get_folders_names(TALLOC_CTX *mem_ctx, struct openchangedb_c
 }
 // ^ openchangedb -------------------------------------------------------------
 
+// v openchangedb replica mapping ---------------------------------------------
+
+static void replica_mapping_add_pair(struct tdb_context *tdb, const struct GUID *guid, uint16_t replid)
+{
+	TDB_DATA	guid_key;
+	TDB_DATA	replid_key;
+	void		*mem_ctx;
+
+	mem_ctx = talloc_new(NULL);
+
+	/* GUID to ReplID */
+	guid_key.dptr = (unsigned char *) GUID_string(mem_ctx, guid);
+	guid_key.dsize = strlen((const char *) guid_key.dptr);
+
+	replid_key.dptr = (unsigned char *) talloc_asprintf(mem_ctx, "0x%.4x", replid);
+	replid_key.dsize = strlen((const char *) replid_key.dptr);
+
+	tdb_store(tdb, guid_key, replid_key, TDB_INSERT);
+	tdb_store(tdb, replid_key, guid_key, TDB_INSERT);
```

Comment:
these 2 tdb_store are visibly related and index the same data with different keys. I would therefore prefer these calls to be "batched": store if both succeed, otherwise rollback. This would reduce paths to data inconsistency.